src/main/java/de/tum/in/www1/artemis/web/rest/ResultResource.java

```diff
@@ -402,14 +417,7 @@ else if (participation instanceof ProgrammingExerciseParticipation) {
     @PreAuthorize("hasRole('TA')")
     public ResponseEntity<Void> deleteResult(@PathVariable Long participationId, @PathVariable Long resultId) {
         log.debug("REST request to delete Result : {}", resultId);
-        Result result = resultRepository.findByIdElseThrow(resultId);
-        Participation participation = result.getParticipation();
-        if (!participation.getId().equals(participationId)) {
-            return badRequest("participationId", "400",
-                    "participationId of the path doesnt match the participationId of the participation corresponding to the result " + resultId + " !");
-        }
-        Course course = participation.getExercise().getCourseViaExerciseGroupOrCourseMember();
-        authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.TEACHING_ASSISTANT, course, null);
+        Result result = getResultForParticipationAndCheckAccess(participationId, resultId, Role.TEACHING_ASSISTANT);
```

Comment:
Coday warning: Result is unused. Is this on purpose to only check the access? If yes I think you should just call the method.